index.js

```diff
@@ -25,10 +25,11 @@ var KNOWN_STATICS = {
 };
 
 module.exports = function hoistNonReactStatics(targetComponent, sourceComponent) {
-    var keys = Object.getOwnPropertyNames(sourceComponent);
-    for (var i=0; i<keys.length; ++i) {
-        if (!REACT_STATICS[keys[i]] && !KNOWN_STATICS[keys[i]]) {
-            targetComponent[keys[i]] = sourceComponent[keys[i]];
+    var sourceKeys = Object.getOwnPropertyNames(sourceComponent),
+        targetKeys = Object.getOwnPropertyNames(targetComponent);
+    for (var i=0; i<sourceKeys.length; ++i) {
+        if (!REACT_STATICS[sourceKeys[i]] && !KNOWN_STATICS[sourceKeys[i]] && targetKeys.indexOf(sourceKeys[i]) === -1) {
```

Comment:
No. In case of `__reactPatchProxy`, `hasOwnProperty` is returning false. I wonder if thats because of the way it is defined (https://github.com/gaearon/react-proxy/blob/v2.0.1/src/createClassProxy.js#L166-L171):

```js
Object.defineProperty(ProxyComponent, '__reactPatchProxy', {
    configurable: false,
    writable: false,
    enumerable: false,
    value: proxy
});
```